_posts/2018-06-13-rpm-ostree-automatic-updates.md

```diff
@@ -0,0 +1,178 @@
+```
+
+We can turn on the update checking by editing `/etc/rpm-ostreed.conf`
+and uncommenting the `AutomaticUpdatePolicy` and `IdleExitTimeout`
+lines.  We want to change the `AutomaticUpdatePolicy` to `check`.
+
+```
+# cat /etc/rpm-ostreed.conf
+# Entries in this file show the compile time defaults.
+# You can change settings by editing this file.
+# For option meanings, see rpm-ostreed.conf(5).
+
+[Daemon]
+AutomaticUpdatePolicy=check
+IdleExitTimeout=60
+```
+
+After configuring the policy, we need to reload the `rpm-ostreed` service
+and enable the systemd timer.  Then we can check the output of
+`rpm-ostree status` to confirm that the policy is in place and the timer
+is active.
+
+```
+# systemctl reload rpm-ostreed
+# systemctl enable rpm-ostreed-automatic.timer --now
+Created symlink /etc/systemd/system/timers.target.wants/rpm-ostreed-automatic.timer → /usr/lib/systemd/system/rpm-ostreed-automatic.timer.
+# rpm-ostree status
+State: idle; auto updates enabled (check; last run 195ms ago)
+Deployments:
+● ostree://fedora-workstation:fedora/28/x86_64/workstation
+                   Version: 28.20180601.0 (2018-06-01 11:53:40)
+                    Commit: 6067049927936b5bf41695263c43f37238603b858506a45c98316501c94841b0
+              GPGSignature: Valid signature by 128CF232A9371991C8A65695E08E7E629DB62FB1
+
+Available update:
+        Version: 28.20180613.0 (2018-06-13 14:05:43)
+         Commit: 106280bdaf280e5c1a2e803f9bb58ce58ab3043ebc69de12c39e030218da2da6
+   GPGSignature: Valid signature by 128CF232A9371991C8A65695E08E7E629DB62FB1
+           Diff: 79 upgraded, 1 removed, 1 added
+```
+
+You can see in the output that the timer was activated and the corresponding
+service was run `195ms` ago.  The service found an update was available and
+the output from `rpm-ostree status` nicely displays information about the
+available upgrade.  A great feature of this new status is that it will
+display any CVEs that are addressed by an available update; in this case
+there are no CVEs addressed, though.  Having this kind of information
+available in the status output allows admins to make better decisions about
+taking the downtime hit to reboot a host to apply an upgrade.
+
+## Automatic Downloading and Staging of Updates
+
+Now let's take the next steps and enable the automatic download and staging
+of the upgrades in the background.  This feature is still considered
+experimental, so you are accepting some amount of risk by using it.  However,
+by enabling the feature on your host and [reporting issues](https://github.com/projectatomic/rpm-ostree/issues/),
+you are helping to stabilize this feature for production-grade deployments.
+
+Again, we configure this by changing the `/etc/rpm-ostreed.conf` file.  This
+time we configure the `AutomaticUpdatePolicy` to a new value of `ex-stage`.
+Additionally, we populate the config file with a new section called
+`[Experimental]` where we can enable the `StageDeployments` setting.  And
```

Comment:
Just a clarification here; you only need to do one or the other. Using `ex-stage` will use the new staging feature only for automatic updates. Using `StageDeployments=yes` will use it for *all* update operations (e.g. `rpm-ostree upgrade`, but also automatic deployments).

Reply:
What's the greater impact of using `StageDeployments=yes` in terms of 'all update operations'?  Does this tie into doing the `/etc/` merge right before reboot?